metadata.json

```diff
@@ -2,7 +2,7 @@
   "author": "Vox Pupuli",
   "license": "Apache-2.0",
   "name": "puppet-r10k",
-  "version": "10.2.1-rc0",
+  "version": "10.2.2-rc0",
```

Comment:
change it to 10.3.0, then run the changelog rale task again. that should generate the correct changelog.md.

Reply:
What should I have done? The instructions are not quite so clear. Perhaps you could write each step you do and I can verify it here and we can update the VP docs together?

Reply:
CHANGELOG did not change btw... these are unreleased changes, so that makes sense to me.

```
$ bundle exec rake changelog

Found 89 tags
Fetching tags dates: 89/89
Sorting tags...
git diffFetching issues... 100/600
Received issues: 553
Pull Request count: 402
Filtered pull requests: 271
Filtered issues: 134
Fetching events for issues and PR: 405
Fetching closed dates for issues: 405/405
Warning: PR 33 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 29 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 28 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Associating PRs with tags: 271/271
Generating entry...
Done!
```

Reply:
did you check https://voxpupuli.org/docs/releasing_version/? 

I think the following sentence is a bit missleading:
> We can generate the changelog after updating the metadata.json with a rake task

and it should probably be:
> We can generate the changelog, after updating the metadata.json, with a rake task

And afterwards elaborate a bit on the version.

Some background on the release process: At the bottom we have the following line:

> bump the current version to the next PATCH version and add -rc0 to the end

Our `release` rake tasks checks if the version in metadata.json isn't a release candidate (no `-rc*`). If it's an RC, the release is aborted. If it's no RC, the release is done and afterwards the version is increased to the next patch version and -rc0 is added. The idea is that you can run the release task multiple times, but it doesn't try to actually do multiple releases for the same version, it's a safety check. Because of that, our metadata.json files in master always have the `-rc0` suffix. This needs to be removed in a release PR. And the version often needs to be adjusted properly to a major or minor version. This is documented a bit in this section:

> The changelog generator checks for certain labels on closed issues and PRs since the last release and groups them together. If the changes were neither backwards-incompatible nor only bug fixes, make a minor release. Check the generated diff for the CHANGELOG.md. If your chosen release version doesn’t match the generated changelog, update metadata.json and run the changelog task again.